src/sortStudents.ts

```diff
@@ -1,16 +1,63 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
 // create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+const calculateAverageGrades = (grades: number[]): number => {
+  return grades.reduce((acc, num) => acc + num, 0) / grades.length;
+};
 
-export function sortStudents(students, sortBy, order) {
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
   // write your function
+  const studentsToSort = [...students];
+
+  return studentsToSort.sort((a, b) => {
+    switch (sortBy) {
+      case SortType.Married:
+      case SortType.Age:
+        if (order === 'desc') {
+          return +b[sortBy] - +a[sortBy];
+        }
+
+        return +a[sortBy] - +b[sortBy];
```

Comment:
it will be great use ternary here, and in other cases
```suggestion
        order === 'desc'
         ?  +b[sortBy] - +a[sortBy];
         :  +a[sortBy] - +b[sortBy];
```